source/slang/slang-ir-any-value-marshalling.cpp

```diff
@@ -719,10 +733,34 @@ namespace Slang
             }
             return offset;
         }
+        case kIROp_AnyValueType:
+        {
+            auto anyValueType = cast<IRAnyValueType>(type);
+            return alignUp(offset, 4) + (SlangInt)getIntVal(anyValueType->getSize());
+        }
+        case kIROp_TupleType:
+        {
+            auto tupleType = cast<IRTupleType>(type);
+            for (UInt i = 0; i < tupleType->getOperandCount(); i++)
+            {
+                auto elementType = tupleType->getOperand(i);
+                offset = _getAnyValueSizeRaw((IRType*)elementType, offset);
+                if (offset < 0) return offset;
+            }
+            return offset;
+        }
+        case kIROp_WitnessTableType:
+        case kIROp_WitnessTableIDType:
+        case kIROp_RTTIHandleType:
+        {
+            return alignUp(offset, 4) + 8;
+        }
         case kIROp_InterfaceType:
         {
-            // TODO: implement anyValue packing for interface types.
-            return -1;
+            auto interfaceType = cast<IRInterfaceType>(type);
+            auto size = SharedGenericsLoweringContext::getInterfaceAnyValueSize(interfaceType, interfaceType->sourceLoc);
+            size += 16;
```

Comment:
Why 16? The 4's I'm presuming because it's an int/uint. Might we want to make these magic numbers an enum or constant value, such that can have a name saying what it is, and allowing for easier future changes.